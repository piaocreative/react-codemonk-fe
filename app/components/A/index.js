/**
 * A link to a certain page, an anchor tag
 */

import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { primaryNew, primaryNewHover, white } from 'themes/variables';

export const A = styled.a`
  color: rgb(${primaryNew});
  font-family: 'GT-Walsheim-Pro-Regular';
  font-size: 16px;
  text-decoration: none;

  &.sm {
    font-size: 14px;
    letter-spacing: -0.02em;
  }

  &[disabled] {
    pointer-events: none;
  }
  &.pdf-icon {
    &[disabled] {
      opacity: 0.5;
    }
  }
  &:hover {
    color: rgb(${primaryNew});
  }
`;

export const LinkMod = styled(Link)`
  color: rgb(${white});
  border-radius: 34px;
  min-width: 190px;
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  padding: 6px 45px;
  text-decoration: none;
  -webkit-font-smoothing: antialiased;
  transition: background 0.3s ease;
  border: none;
  text-align: center;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-weight: bold;
  font-size: 16px;
  transition: all 0.3s ease;

  &.btn-primary {
    background: rgb(${primaryNew});
    &:disabled {
      &:hover {
        background: rgb(${primaryNew});
      }
    }
    &:hover {
      background: rgba(${primaryNewHover});
      text-decoration: none;
    }
    &:active {
      background-color: rgb(${primaryNew});
    }
  }
`;
