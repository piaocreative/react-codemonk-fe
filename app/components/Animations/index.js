import { keyframes } from 'styled-components';

export const Rotate = keyframes`
  0% {
    transform: translateY(-50%) rotate(0deg);
  }

  100% {
    transform: translateY(-50%) rotate(360deg);
  }
`;
Rotate.displayName = 'rotate';
