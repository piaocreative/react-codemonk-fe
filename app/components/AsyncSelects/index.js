/**
 * over ride the react-select and bind with redux-form so we can do a validation of select box.
 * @author Innovify
 *
 */
import React from 'react';
import { FormGroup } from 'reactstrap';
import AsyncSelect from 'react-select/async';
import AsyncCreatableSelect from 'react-select/async-creatable';
import { defaultProps, propTypes } from 'containers/proptypes';
import { VALIDATION } from 'utils/constants';
import { primaryNew } from 'themes/variables';
import { SelectContainer, selectStyle } from '../Selects/style';

import filterOptions from '../Selects/filterOptions';
import { ValidationMessage } from '../Input';
import { DropdownIndicator } from '../DropDownIndicator';

export class AsyncSelects extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      defaultValue: props.defaultValue,
      hasMultiError: false,
    };
  }

  /**
   * call when component will receive Params.
   */
  componentWillReceiveProps(nextProps) {
    const { defaultValue } = this.props;
    if (nextProps.defaultValue !== defaultValue) {
      this.setDefaultValue(nextProps.defaultValue);
    }
  }

  /**
   * onChange call on data value selected by user
   * @param {object} a json of data
   * @author Innovify
   */
  onChange = selectedOption => {
    const { input, multi, length, clearable, defaultValue } = this.props;
    if (input.onChange && selectedOption && (selectedOption.value || selectedOption.length)) {
      if (multi) {
        if (selectedOption.length <= length) {
          this.setState({ defaultValue: selectedOption });
          input.onChange(selectedOption);
          this.setState({ hasMultiError: false });
        } else {
          this.setState({ hasMultiError: true });
        }
      } else {
        this.setState({ defaultValue: selectedOption });
        input.onChange(selectedOption);
      }
    } else if (multi) {
      this.setState({ defaultValue: selectedOption });
      input.value = '';
      input.onChange(selectedOption);
    } else if (clearable) {
      this.setState({ defaultValue: '' });
      input.onChange('');
    } else {
      const value = defaultValue;
      this.setState({ defaultValue: value });
      input.value = '';
    }
  };
  /**
   * onChange call on data value selected by user
   *
   * @param {object} a json of data
   * @author Innovify
   * @commented we do not need it, issue on multiple select on same form
   */

  onBlur = () => {
    const { input } = this.props;
    input.onBlur(input.value);
  };

  /**
   * onChange call on data value selected by user
   * @param {object} a json of data
   * @author Innovify
   */
  setDefaultValue = defaultValue => {
    this.setState({ defaultValue });
  };

  noOptionMessage = () => {
    const { defaultOptions } = this.props;
    const { enteredCharCount } = this.state;
    if (enteredCharCount > 1 && defaultOptions.length === 0) {
      return 'No Options';
    }
    return '';
  };

  render() {
    const {
      multi,
      placeHolder,
      length,
      meta: { touched, error, warning },
      input,
      className,
      fullWidthOption,
      loadOptions,
      defaultOptions,
      handleChange,
      disable = false,
      creatable = false,
    } = this.props;
    const { defaultValue } = this.state;
    const { hasMultiError } = this.state;
    const isValid = touched && !defaultValue && error;
    const warnging = hasMultiError ? 'has-warning' : '';
    const errorClass = isValid ? 'has-error' : '';
    const success = touched && !error && !hasMultiError ? 'success' : '';
    const errors = isValid ? 'has-error' : success;

    const selectProps = {
      cacheOptions: true,
      defaultOptions,
      loadOptions,
      onChange: handleChange,
      defaultValue,
      onBlur: () => input.onBlur(),
      value: defaultValue,
      placeholder: placeHolder,
      isClearable: true,
      isDisabled: disable,
      components: { DropdownIndicator },
      filterOptions,

      styles: selectStyle(fullWidthOption),
      noOptionsMessage: this.noOptionMessage,
      onInputChange: value => {
        this.setState({ enteredCharCount: value.length });
      },
      theme: theme => ({
        ...theme,
        borderRadius: 0,
        colors: {
          ...theme.colors,
          primary25: `rgba(${primaryNew}, 0.1)`,
          primary50: `rgba(${primaryNew}, 0.1)`,
          primary: `rgb(${primaryNew})`,
        },
      }),
    };

    const syncSelect = creatable ? (
      <AsyncCreatableSelect {...selectProps} {...this.props} className="custom-selectbox" />
    ) : (
      <AsyncSelect {...selectProps} {...this.props} className="custom-selectbox" />
    );

    return (
      <FormGroup className={`${warnging} ${errorClass}`}>
        <div className={`${errors} form-control-validated`}>
          <SelectContainer className={className}>{syncSelect}</SelectContainer>
          <ValidationMessage>{touched && !defaultValue && (error || warning)}</ValidationMessage>
          {multi && hasMultiError && (
            <ValidationMessage>
              {VALIDATION.MAX_LENGTH} {length}
            </ValidationMessage>
          )}
        </div>
      </FormGroup>
    );
  }
}

AsyncSelects.defaultProps = defaultProps;
AsyncSelects.propTypes = propTypes;

export default AsyncSelects;
