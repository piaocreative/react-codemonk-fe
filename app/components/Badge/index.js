import styled from 'styled-components';
import { primaryDarkNew, danger, orange, success, primaryGradientRight, primaryNew } from 'themes/variables';

const Badge = styled.span`
  color: rgb(${primaryDarkNew});
  font-family: 'GT-Walsheim-Pro-Regular';
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  font-size: 16px;
  line-height: 18px;
  padding: 6px 15px;
  border-radius: 15px;

  &.new-badge {
    font-size: 10px;
    font-family: 'GT-Walsheim-Pro-Regular';
    padding: 5px;
    position: absolute;
    right: -35px;
    top: -4px;
    z-index: 9;
    border-radius: 10px;

    @media (max-width: 1199px) {
      right: -7px;
      top: -12px;
      font-size: 8px;
    }
  }

  &.light {
    background-color: rgba(${primaryDarkNew}, 0.1);
    color: rgb(${primaryDarkNew});
  }
  &.success {
    background-color: rgb(38, 167, 61, 0.1);
    color: ${success};
  }
  &.warning {
    background-color: rgba(222, 189, 24, 0.1);
    color: rgb(222, 189, 24);
  }
  &.danger {
    background-color: rgba(226, 62, 71, 0.1);
    color: ${danger};
  }
  &.alert {
    background-color: rgba(${orange}, 0.1);
    color: rgb(${orange});
  }
  &.primary {
    background: ${primaryGradientRight};
  }
  &.secondary {
    background: rgb(${primaryNew}, 0.1);
  }
  &.badge-sm {
    font-size: 14px;
    line-height: 16px;
    padding: 4px 10px;
    border-radius: 12px;
  }
  &.badge-status {
    border-radius: 12px;
    margin-bottom: 0px;
  }
`;
export default Badge;
