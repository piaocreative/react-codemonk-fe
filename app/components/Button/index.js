/**
 *
 * Button.js
 *
 * A common button, if you pass it a prop "route" it'll render a link to a react-router route
 * otherwise it'll render a link with an onclick
 */

import React, { Children } from 'react';
import PropTypes from 'prop-types';
import StyledButton from './StyledButton';
import StyledLink from './StyledLink';

function Button(props) {
  let button = (
    <StyledButton className={props.btnClassName} {...props}>
      {Children.toArray(props.children)}
    </StyledButton>
  );
  // Render an anchor tag
  if (props.buttonAsLink) {
    button = (
      <StyledLink to={props.href} className={props.btnClassName}>
        {Children.toArray(props.children)}
      </StyledLink>
    );
  }

  return button;
}

Button.propTypes = {
  handleRoute: PropTypes.func,
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  background: PropTypes.string,
  color: PropTypes.string,
  btnClassName: PropTypes.string,
};

export default Button;
