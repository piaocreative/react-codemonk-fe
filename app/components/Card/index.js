import styled from 'styled-components';
import { white, primaryGradientRight } from 'themes/variables';
import media from 'themes/media';

const Card = styled.div`
  background: rgb(${white});
  padding: 30px 25px;
  border-radius: 5px;

  ${media.large`
		padding: 50px;
	`};

  .auth-layout > & {
    flex: 1 1 0%;
    height: 100%;
  }

  &.pagination-block {
    margin-top: 5px;
    padding: 20px 40px !important;
  }
  &.table-header {
    padding: 30px 40px;
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;

    & .form-group {
      margin-bottom: 0;
      width: 200px;
    }
  }
  &.bg-secondary-gradient {
    background: ${primaryGradientRight};
  }
  &.talent-card {
    :first-child {
      margin-top: 20px;
    }
  }
`;

export default Card;
