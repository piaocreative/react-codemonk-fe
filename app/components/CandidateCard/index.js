/* eslint-disable prettier/prettier */
/*
 * CandidateCard
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import get from 'lodash/get';
import { Badge, P, Card } from 'components';
import { getLabel, getArrayLabels, getTimzoneOffest, languageLabel } from 'containers/MyProfilePage/components/utils';
import { yearsOfExperienceArray, timeXZone, teamPreferenceArray } from 'containers/App/constants';
import { PreferencesList } from 'containers/MyProfilePage/styles';
import messages from './messages';

function CandidateCard(props) {
  const { briefData } = props;

  const handleActiveField = (value, type) => {
    const userData = get(briefData, 'userData', {});
    const userDataSkill = get(userData, 'skills', []);
    let isActive = 'secondary';
    let matchItem;
    let userTeamTypes = [];
    switch (type) {
      case 'expertise':
        matchItem = getLabel(get(userData, 'yearsOfExperience', ''), yearsOfExperienceArray, 'label') === value && true;
        break;
      case 'skills':
      case 'softSkills':
        matchItem = userDataSkill.find(item => item.name === value);
        break;
      case 'language':
        matchItem = get(userData, type, []).find(item => languageLabel(item) === value);
        break;
      case 'timeZone':
        matchItem = get(userData, type) === value && true;
        break;
      case 'industries':
        matchItem = get(userData, type, []).find(item => item === value);
        break;
      case 'teamPreference':
        userTeamTypes = getArrayLabels(get(userData, type, []), teamPreferenceArray);
        matchItem = userTeamTypes.find(item => item === value);
        break;
      case 'certificateDetails':
        matchItem = get(userData, type, []).find(item => item.name === value);
        break;
      default:
        isActive = 'secondary';
        break;
    }
    isActive = matchItem ? 'success' : 'secondary';

    return isActive;
  };

  return (
    <>
      <Card className="pt-4">
        <P className="p20">
          <FormattedMessage {...messages.title} />
        </P>
        <div className="mt-4 pt-1">
          <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
            <FormattedMessage {...messages.seniority} />
          </P>
          <div className="inline-list">
            <Badge className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(get(briefData, 'expertise'), 'expertise')}`}>
              {get(briefData, 'expertise')}
            </Badge>
          </div>
        </div>
        {get(briefData, 'skills') && get(briefData, 'skills').length > 0 && (
          <>
            <hr className="mt-3 mb-3" />
            <div>
              <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                <FormattedMessage {...messages.skillFrameworks} />
              </P>
              <PreferencesList className="inline-list">
                {get(briefData, 'skills').map(brief => (
                  <Badge className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(brief, 'skills')}`} key={brief}>
                    {brief}
                  </Badge>
                ))}
              </PreferencesList>
            </div>
          </>
        )}

        {get(briefData, 'softSkills') && get(briefData, 'softSkills').length > 0 && (
          <>
            <hr className="mt-3 mb-3" />
            <div>
              <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                <FormattedMessage {...messages.softSkills} />
              </P>
              <div className="inline-list">
                {get(briefData, 'softSkills').map(brief => (
                  <Badge className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(brief, 'softSkills')}`} key={brief}>
                    {brief}
                  </Badge>
                ))}
              </div>
            </div>
          </>
        )}

        {get(briefData, 'certifications') && get(briefData, 'certifications').length > 0 && (
          <>
            <hr className="mt-3 mb-3" />
            <div>
              <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                <FormattedMessage {...messages.certifications} />
              </P>
              <div className="inline-list">
                {get(briefData, 'certifications').map(brief => (
                  <Badge
                    className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(brief, 'certificateDetails')}`}
                    key={brief}
                  >
                    {brief}
                  </Badge>
                ))}
              </div>
            </div>
          </>
        )}

        {get(briefData, 'languages') && get(briefData, 'languages').length > 0 && (
          <>
            <hr className="mt-3 mb-3" />
            <div>
              <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                <FormattedMessage {...messages.language} />
              </P>
              <div className="inline-list">
                {get(briefData, 'languages').map(brief => (
                  <Badge className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(brief, 'language')}`} key={brief}>
                    {brief}
                  </Badge>
                ))}
              </div>
            </div>
          </>
        )}

        {get(briefData, 'timeZone') && (
          <>
            <hr className="mt-3 mb-3" />
            <div>
              <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                <FormattedMessage {...messages.timezone} />
              </P>
              <div className="inline-list">
                <Badge className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(get(briefData, 'timeZone'), 'timeZone')}`}>
                  {getTimzoneOffest(timeXZone, 'name', get(briefData, 'timeZone'))} ({get(briefData, 'timeZone')})
                </Badge>
              </div>
            </div>
          </>
        )}

        {get(briefData, 'teamPrefArray') &&
          get(briefData, 'teamPrefArray').length > 0 &&
          !(get(briefData, 'teamPrefArray').length === 1 && get(briefData, 'teamPrefArray').includes(undefined)) && (
            <>
              <hr className="mt-3 mb-3" />
              <div>
                <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                  <FormattedMessage {...messages.teamWork} />
                </P>
                <div className="inline-list">
                  {get(briefData, 'teamPrefArray').map(
                    brief =>
                      brief && (
                        <Badge
                          className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(brief, 'teamPreference')}`}
                          key={brief}
                        >
                          {brief}
                        </Badge>
                      ),
                  )}
                </div>
              </div>
            </>
          )}

        {get(briefData, 'industry') && (
          <>
            <hr className="mt-3 mb-3" />
            <div>
              <P className="p14 mb-1" fontFamily="GT-Walsheim-Pro-Medium">
                <FormattedMessage {...messages.industry} />
              </P>
              <div className="inline-list">
                <Badge className={`text-ellipsis mw-100 mt-2 me-2 badge-sm ${handleActiveField(get(briefData, 'industry'), 'industries')}`}>
                  {get(briefData, 'industry')}
                </Badge>
              </div>
            </div>
          </>
        )}
      </Card>
    </>
  );
}

CandidateCard.defaultProps = {
  briefData: {},
};

CandidateCard.propTypes = {
  briefData: PropTypes.object,
};

export default CandidateCard;
