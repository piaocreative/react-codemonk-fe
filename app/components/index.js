import { Input } from 'reactstrap';
import ContainerMod from './ContainerMod';
import ProgressMod from './ProgressMod';
import Card from './Card';
import H1 from './H1';
import H2 from './H2';
import H3 from './H3';
import H4 from './H4';
import H5 from './H5';
import H6 from './H6';
import Badge from './Badge';
import UserInfoList from './UserInfoList';
import ImageGrid from './ImageGrid';
import LinkButtonMod from './LinkButtonMod';
import FormWrapper from './FormWrapper';
import FormLabel from './Label';
import Button from './Button';
import PrivateGrid from './PrivateGrid';
import Ratings from './Ratings';
import { FormControl, FormControlWrapper, FormControlSelect, ValidationMessage, CheckBoxContainer } from './Input';
import EditorWrapper from './EditorWrapper';
import SelectTags from './SelectTags';
import P from './P';
import SVGIcon from './SVGIcon';
import { OnboardingForm, OnBoardingFormBody } from './OnboardingForm';
import ToastifyMessage from './ToastifyMessage';
import DatePickers from './DatePickers';
import Selects from './Selects';
import { A } from './A';
import UserList from './UserList';
import StyledPagination from './StyledPagination';

export {
  UserInfoList,
  ImageGrid,
  PrivateGrid,
  ContainerMod,
  ProgressMod,
  Card,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Badge,
  LinkButtonMod,
  FormWrapper,
  FormLabel,
  Button,
  Ratings,
  FormControl,
  FormControlWrapper,
  FormControlSelect,
  ValidationMessage,
  EditorWrapper,
  SelectTags,
  P,
  SVGIcon,
  OnboardingForm,
  OnBoardingFormBody,
  Input,
  CheckBoxContainer,
  ToastifyMessage,
  DatePickers,
  Selects,
  A,
  UserList,
  StyledPagination,
};
