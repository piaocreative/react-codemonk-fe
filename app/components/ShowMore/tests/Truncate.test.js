import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import { LinkButtonMod } from 'components';

import Truncate from '../Truncate';

const props = {
  lines: 3,
  more: 'more',
  less: '',
  anchorClass: 'links',
  onClick: jest.fn(),
  expanded: false,
};

describe('<Truncate />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<Truncate />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});

const testMessage =
  'Test Message Lorem ipsum dolor sit amet, <a href="https://www.google.com/">Google link</a> consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.';

describe(' <Truncate />', () => {
  const renderComponent = (props = {}) => render(<Truncate {...props}>{testMessage}</Truncate>);

  it('should render an <Truncate /> tag', () => {
    const { container } = renderComponent();
    expect(container.querySelector('span')).not.toBeNull();
  });
});

const getWrapper = () => shallow(<Truncate {...props} />);
const getWrapper2 = () =>
  shallow(
    <Truncate
      width={0}
      lines={3}
      ellipsis={
        <span>
          ...
          <LinkButtonMod className="mb-0 ms-1" color="link" onClick={jest.fn()}>
            more
          </LinkButtonMod>
        </span>
      }
      onTruncate={jest.fn()}
      ref={ref => (this.truncateRef = ref)}
      {...props}
    >
      {testMessage}
    </Truncate>,
  );
const getInstance = () => getWrapper().instance();
const getInstance2 = () => getWrapper2().instance();
describe('test functions', () => {
  getWrapper();

  test('Testing restoreReplacedLinks', () => {
    const restoreReplacedLinks = jest.spyOn(getInstance(), 'restoreReplacedLinks');
    restoreReplacedLinks(testMessage);
    expect(restoreReplacedLinks).toHaveBeenCalledTimes(1);
  });

  test('Testing renderLine with if', () => {
    const renderLine = jest.spyOn(getInstance2(), 'renderLine');
    renderLine(2, 2, [1, 2, 3]);
    expect(renderLine).toHaveBeenCalledTimes(1);
  });
  test('Testing renderLine with else', () => {
    const renderLine = jest.spyOn(getInstance2(), 'renderLine');
    renderLine(2, 1, []);
    expect(renderLine).toHaveBeenCalledTimes(1);
  });
  test('Testing renderLine with else- nested else', () => {
    const renderLine = jest.spyOn(getInstance2(), 'renderLine');
    renderLine(null, 1, []);
    expect(renderLine).toHaveBeenCalledTimes(1);
  });

  test('Testing innerText ', () => {
    const innerText = jest.spyOn(getInstance2(), 'innerText');
    innerText(<span>{testMessage}</span>);
    expect(innerText).toHaveBeenCalledTimes(1);
  });

  test('Testing onResize ', () => {
    const onResize = jest.spyOn(getInstance2(), 'onResize');
    onResize();
    expect(onResize).toHaveBeenCalledTimes(1);
  });

  test('Testing onTruncate ', () => {
    const onTruncate = jest.spyOn(getInstance2(), 'onTruncate');
    onTruncate(true);
    expect(onTruncate).toHaveBeenCalledTimes(1);
  });

  test('Testing ellipsisWidth ', () => {
    const ellipsisWidth = jest.spyOn(getInstance2(), 'ellipsisWidth');
    ellipsisWidth(<span>{testMessage}</span>);
    expect(ellipsisWidth).toHaveBeenCalledTimes(1);
  });

  test('Testing trimRight ', () => {
    const trimRight = jest.spyOn(getInstance2(), 'trimRight');
    trimRight('hello');
    expect(trimRight).toHaveBeenCalledTimes(1);
  });

  test('Testing createMarkup ', () => {
    const createMarkup = jest.spyOn(getInstance2(), 'createMarkup');
    createMarkup('hello');
    expect(createMarkup).toHaveBeenCalledTimes(1);
  });
});
