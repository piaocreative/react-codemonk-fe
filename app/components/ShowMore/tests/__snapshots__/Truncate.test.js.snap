// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Truncate /> should match the snapshot 1`] = `
<span
  width={0}
>
  <span />
  <span>
    
  </span>
  <span
    style={
      Object {
        "left": 0,
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
      }
    }
  >
    â€¦
  </span>
</span>
`;
